Z-Indexing revisited

Alex-Weatherhead opened the pull request:
Due to the way the markers were being sorted and placed, two issues had arisen: (1) when calling closeMetadata(), the markers were not added back in the sorted order, which was causing problems; (2) I had accidentally used the index of the sorted markers as the currentMetadata, but I needed to use the original indices, because of how the cards are created.

Alex-Weatherhead committed:
Added Z-Indexing to markers based on BB area.

Leaflet has some inbuilt z-indexing stuff, but it didn't seem to fit our needs. So pretty much, I just sorted the markers by the area of their bounding boxes, and then whenever something gets toggled, I remove all the markers, and then add them all back biggest to smallest. This way, the smallest markers are always clickable.

Alex-Weatherhead committed:
Add files via upload

Alex-Weatherhead committed:
Fixed Z-Indexing of markers

Due to the way the markers were being sorted and placed, two issues had arisen: (1) when calling closeMetadata(), the markers were not added back in the sorted order, which was causing problems; (2) I had accidentally used the index of the sorted markers as the currentMetadata, but I needed to use the original indices, because of how the cards are created.

gauchm commented on index.js:
```diff
@@ -61,6 +61,8 @@ function hashChanged() {
     if (target !== "") {
         for (let i = 0; i < datasets.length; i++) {
+            //let marker_size_and_index_pair = marker_size_and_index_pairs[i];
+            //let marker_index = marker_size_and_index_pair[1];
```
Can we remove these commented lines?

Alex-Weatherhead committed:
Added z-indexing to the most recent version.

Had to revisit z-indexing in the newest code-base, as several changes were no longer compatible.

Alex-Weatherhead commented:
I'm actually going to close this pull request and open a new one.
